### ISSUE TITLE
DRAM-v.py annotate  KeyError: 'K05663'

### ISSUE THREAD

**Issue by Shild94** [WrightonLabCSU/DRAM#72]
Hi,
Thanks for providing a very powerful tool. I got an error when using DRAM-v.py annotate. 
My command is:
DRAM-v.py annotate -i   vs_out/for-dramv/final-viral-combined-for-dramv.fa -v   vs_out/for-dramv/viral-affi-contigs-for-dramv.tab   -o  test

The log file shows:

1:20:10.376907: Annotating k141_1108119__full-cat_1
1:20:10.511786: Turning genes from prodigal to mmseqs2 db
1:20:12.205664: Getting hits from kofam
Traceback (most recent call last):
  File "/biostack/bioconda/envs/DRAM/lib/python3.9/site-packages/pandas/core/indexes/base.py", line 3080, in get_loc
    return self._engine.get_loc(casted_key)
  File "pandas/_libs/index.pyx", line 70, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/index.pyx", line 101, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/hashtable_class_helper.pxi", line 4554, in pandas._libs.hashtable.PyObjectHashTable.get_item
  File "pandas/_libs/hashtable_class_helper.pxi", line 4562, in pandas._libs.hashtable.PyObjectHashTable.get_item
KeyError: 'K05663'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/biostack/bioconda/envs/DRAM/bin/DRAM-v.py", line 140, in <module>
    args.func(**args_dict)
  File "/biostack/bioconda/envs/DRAM/lib/python3.9/site-packages/mag_annotator/annotate_vgfs.py", line 403, in annotate_vgfs
    annotations = annotate_fastas(contig_locs, output_dir, db_locs_anno, db_handler, min_contig_size, prodigal_mode,
  File "/biostack/bioconda/envs/DRAM/lib/python3.9/site-packages/mag_annotator/annotate_bins.py", line 923, in annotate_fastas
    annotations_list.append(annotate_fasta(fasta_loc, fasta_name, fasta_dir, db_locs, db_handler, min_contig_size,
  File "/biostack/bioconda/envs/DRAM/lib/python3.9/site-packages/mag_annotator/annotate_bins.py", line 824, in annotate_fasta
    annotations = annotate_orfs(gene_faa, db_locs, tmp_dir, start_time, db_handler, custom_db_locs, bit_score_threshold,
  File "/biostack/bioconda/envs/DRAM/lib/python3.9/site-packages/mag_annotator/annotate_bins.py", line 735, in annotate_orfs
    annotation_list.append(run_hmmscan_kofam(gene_faa, db_locs['kofam'], tmp_dir,
  File "/biostack/bioconda/envs/DRAM/lib/python3.9/site-packages/mag_annotator/annotate_bins.py", line 242, in run_hmmscan_kofam
    ko_row = ko_list.loc[ko]
  File "/biostack/bioconda/envs/DRAM/lib/python3.9/site-packages/pandas/core/indexing.py", line 895, in __getitem__
    return self._getitem_axis(maybe_callable, axis=axis)
  File "/biostack/bioconda/envs/DRAM/lib/python3.9/site-packages/pandas/core/indexing.py", line 1124, in _getitem_axis
    return self._get_label(key, axis=axis)
  File "/biostack/bioconda/envs/DRAM/lib/python3.9/site-packages/pandas/core/indexing.py", line 1073, in _get_label
    return self.obj.xs(label, axis=axis)
  File "/biostack/bioconda/envs/DRAM/lib/python3.9/site-packages/pandas/core/generic.py", line 3739, in xs
    loc = index.get_loc(key)
  File "/biostack/bioconda/envs/DRAM/lib/python3.9/site-packages/pandas/core/indexes/base.py", line 3082, in get_loc
    raise KeyError(key) from err
KeyError: 'K05663'

The database configurations are:

DRAM-setup.py print_config
KEGG db: None
KOfam db: /biostack/database/DRAM_database/database_files/kofam_profiles.hmm
KOfam KO list: /biostack/database/DRAM_database/database_files/kofam_ko_list.tsv
UniRef db: /biostack/database/DRAM_database/database_files/uniref90.20210323.mmsdb
Pfam db: /biostack/database/DRAM_database/database_files/pfam.mmspro
Pfam hmm dat: /biostack/database/DRAM_database/database_files/Pfam-A.hmm.dat.gz
dbCAN db: /biostack/database/DRAM_database/database_files/dbCAN-HMMdb-V9.txt
dbCAN family activities: /biostack/database/DRAM_database/database_files/CAZyDB.07312019.fam-activities.txt
RefSeq Viral db: /biostack/database/DRAM_database/database_files/refseq_viral.20210327.mmsdb
MEROPS peptidase db: /biostack/database/DRAM_database/database_files/peptidases.20210327.mmsdb
VOGDB db: /biostack/database/DRAM_database/database_files/vog_latest_hmms.txt
VOG annotations: /biostack/database/DRAM_database/database_files/vog_annotations_latest.tsv.gz
Description db: /biostack/database/DRAM_database/database_files/description_db.sqlite
Genome summary form: /biostack/database/DRAM_database/database_files/genome_summary_form.tsv
Module step form: /biostack/database/DRAM_database/database_files/module_step_form.tsv
ETC module database: /biostack/database/DRAM_database/database_files/etc_module_database.tsv
Function heatmap form: /biostack/database/DRAM_database/database_files/function_heatmap_form.tsv
AMG database: /biostack/database/DRAM_database/database_files/amg_database.tsv

Do you have any suggestions to solve this problem？ Thanks in advance.

**Comment by shafferm** [comment]
Can you try grep'ing for that KO in your kofam_ko_list.tsv file? So run `grep K05663 /biostack/database/DRAM_database/database_files/kofam_ko_list.tsv`. Then let me know if that output's anything.

**Comment by Shild94** [comment]
Thanks for replying.
I have looked at that list and found there was no such KO. So I manually added it to the kofam_ko_list.tsv file, and now the "DRAM-v.py annotate" runs perfectly.
However, I encountered another error when running "DRAM-v.py distill", which showed a KeyError: 'amg_flags'. Also, I can not find such column in the annotations.tsv file generated by "DRAM-v.py annotate". I guess it may be the prepare_database issue, because I could not wget the amg_database.tsv, thus downloaded it manually and assigned the location using "set_locations" flag. As I posted last time, print_config does show the location of amg_database.tsv file, but it seems that this database file does not work. What should I do to solve this problem?
Many thanks for your kind help.
